### PR TITLE
SSM schema updates & add update view for SSM

### PR DESCRIPTION
**Pull request by in-op**
Adds UI for updating existing params at a given name. Users can update exiting parameters' type and/or value, delete a param, or even update the name (which is really a delete then a post).

A few minor fixes in here too such as using the correct key for a React list, and fixing a small bug with SlideOver not resetting the toggle to hide SecureStrings by default.

### CODE REVIEW

File: src/js/views/Project/Configuration/AddSSMParam.jsx [hunk 1]
```diff
@@ -29,7 +29,9 @@ function AddSSMParam({ onClose, project, pathPrefix }) {
     if (!saving) return
 
     const data = Object.fromEntries(
-      Object.entries(values).filter(([key, value]) => !!value)
+      Object.entries(values)
+        .filter(([, value]) => !!value)
```
Comment (dave-shawley) [inline src/js/views/Project/Configuration/AddSSMParam.jsx:33]: What is the leading comma for in `([, value])`?

File: src/js/views/Project/Configuration/EditSSMParam.jsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            : response.data
+        )
+      } else {
+        onSuccess()
+      }
+    }
+
+    deleteParams(() => {
+      createParams(() => onSuccess()).catch((error) => onError(error))
+    }).catch((error) => onError(error))
```
Comment (dave-shawley) [inline src/js/views/Project/Configuration/EditSSMParam.jsx:97]: Is there a benefit to reusing the `onSuccess` name and chaining as opposed to having `deleteParams` invoke `createParams` explicitly and `createParams` invoke the outer `onSuccess`?

```javascript
function deleteThenPost() {
  const createParams = async () => {
    //...
    if (!response.success) {
      //...
    } else {
      onSuccess()
    }
  }
  const deleteParams = async() => {
    //...
    if (!response.success) {
      //...
    } else {
      await createParams()
    }
  }
  try {
    await deleteParams()
  } catch (error) {
    onError(error)
  }
}
```

Comment (dave-shawley) [reply src/js/views/Project/Configuration/EditSSMParam.jsx:97]: This confused me until I figured out that `onSuccess` wasn't the outer name even though `onError` did

File: src/js/views/Project/Configuration/EditSSMParam.jsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            <Form.Field
+              title={
+                <Trans
+                  i18nKey={'project.configuration.ssm.environmentType'}
+                  i18n={i18n}
+                  t={t}>
+                  {{ environment }}
+                </Trans>
+              }
+              name={`${environment}`}
```
Comment (dave-shawley) [inline src/js/views/Project/Configuration/EditSSMParam.jsx:191]:
```suggestion
              name={environment}
```
?

File: src/js/views/Project/Configuration/AddSSMParam.jsx (hunk 1 above)
Comment (in-op) [reply src/js/views/Project/Configuration/AddSSMParam.jsx:33]: It's like the equivalent of naming a positional arg `_` in python, it means there is something before the next variable, but it's unused.

File: src/js/views/Project/Configuration/EditSSMParam.jsx (hunk 2 above)
Comment (in-op) [reply src/js/views/Project/Configuration/EditSSMParam.jsx:97]: No, your suggestion is more clear. The only wrinkle is you can't call `await deleteParams()` from the not-async `deleteThenPost`, but that's ok, we can just call `deleteParams().catch((error) => onError(error))` instead.

File: src/js/views/Project/Configuration/EditSSMParam.jsx (hunk 3 above)
Comment (in-op) [reply src/js/views/Project/Configuration/EditSSMParam.jsx:191]: Good catch

**Pull request merged**